src/libs/PolicyDistanceRatesUtils.ts

```diff
@@ -11,14 +11,33 @@ type RateValueForm = typeof ONYXKEYS.FORMS.POLICY_CREATE_DISTANCE_RATE_FORM | ty
 
 type TaxReclaimableForm = typeof ONYXKEYS.FORMS.POLICY_DISTANCE_RATE_TAX_RECLAIMABLE_ON_EDIT_FORM;
 
-function validateRateValue(values: FormOnyxValues<RateValueForm>, currency: string, toLocaleDigit: (arg: string) => string): FormInputErrors<RateValueForm> {
+function validateRateValue(
+    values: FormOnyxValues<RateValueForm>,
+    customUnitRates: Record<string, Rate>,
+    currency: string,
+    toLocaleDigit: (arg: string) => string,
+    currentRateValue?: number,
+): FormInputErrors<RateValueForm> {
     const errors: FormInputErrors<RateValueForm> = {};
     const parsedRate = MoneyRequestUtils.replaceAllDigits(values.rate, toLocaleDigit);
     const decimalSeparator = toLocaleDigit('.');
+    const ratesList = Object.values(customUnitRates);
+    // The following logic replicates the backend's handling of rates:
+    // - Multiply the rate by 100 (CUSTOM_UNIT_RATE_BASE_OFFSET) to scale it, ensuring precision.
+    // - This ensures rates are converted as follows:
+    //   12       -> 1200
+    //   12.1     -> 1210
+    //   12.01    -> 1201
+    //   12.001   -> 1200.1
+    //   12.0001  -> 1200.01
+    // - Using parseFloat and toFixed(10) retains the necessary precision.
+    const convertedRate = parseFloat((Number(values.rate || 0) * CONST.POLICY.CUSTOM_UNIT_RATE_BASE_OFFSET).toFixed(10));
 
     // Allow one more decimal place for accuracy
     const rateValueRegex = RegExp(String.raw`^-?\d{0,8}([${getPermittedDecimalSeparator(decimalSeparator)}]\d{0,${CONST.MAX_TAX_RATE_DECIMAL_PLACES}})?$`, 'i');
-    if (!rateValueRegex.test(parsedRate) || parsedRate === '') {
+    if (ratesList.some((r) => r.rate === convertedRate && !(currentRateValue && currentRateValue === r.rate))) {
```

Comment:
Let's remove !(currentRateValue && currentRateValue === r.rate and filter currentRateValue out from convertedRate

Reply:
Ahh I made a mistake. I mean that we should remove currentRateValue from ratesList and don't need to check ` !(currentRateValue && currentRateValue === r.rate`